gr-azure-software-radio/lib/difi_sink_cpp_impl.cc

```diff
@@ -111,21 +117,55 @@ namespace gr {
       }
       d_out_buf.resize(d_data_len);
     }
+
     template <class T>
     difi_sink_cpp_impl<T>::~difi_sink_cpp_impl()
     {
       close(d_socket);
+      FD_CLR(d_socket,&d_fdset);
     }
 
     template <class T>
     int difi_sink_cpp_impl<T>::work(int noutput_items,
         gr_vector_const_void_star &input_items,
         gr_vector_void_star &output_items)
     {
+      if(d_socket_type == SOCK_STREAM)
+      {
+        socklen_t addr_len = sizeof(d_servaddr);
+        if(getpeername(d_socket, (struct sockaddr*)&d_servaddr, &addr_len) < 0)
+        {
+          close(d_socket);
+          create_tcp_socket();
+
+          if(getpeername(d_socket, (struct sockaddr*)&d_servaddr, &addr_len) < 0)
```

Comment:
Want to make sure I follow this case.
When getpeername() returns a negative does this mean the connect has closed or otherwise indicate an error? If so should we return n_output_items?

Reply:
getpeername returns the address of the socket is connected to, if it returns negative it indicates an error (socket not connected, invalid file descriptor, etc). We could return 0 but if the server never comes back this will slow things down significantly.